File: eFrizer/eFrizer.Model/Reservation/ReservationSearchRequest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eFrizer.Model
{
    public class ReservationSearchRequest
    {
        public int? HairDresserId { get; set; }
        public int? ApplicationUserId { get; set; }
        public DateTime To { get; set; }
        public DateTime From { get; set; }

        public int Day { get; set; }

        public int Month { get; set; }
    }
}

File: eFrizer/flutter_efrizer/lib/models/reservation/reservation_search_request.dart
import 'package:flutter/material.dart';

class ReservationSearchRequest{
  int? applicationuserId;
  int? hairdresserId;
  int? day;
  int? month;

  ReservationSearchRequest({
    this.applicationuserId,
    this.hairdresserId,
    this.day,
    this.month
  });

  factory ReservationSearchRequest.fromJson(Map<String,dynamic> json)
  {
    return ReservationSearchRequest(
      applicationuserId: int.parse(json['clientId'].toString()),
      hairdresserId: int.parse(json['hairDresserId']),
    );
  }
}